packages/background/src/utils/swaps/1inch.ts

```diff
@@ -106,13 +119,129 @@ export interface OneInchSwapRequestResponse {
     toToken: BasicToken;
     fromTokenAmount: string; // Input amount of fromToken in minimal divisible units
     toTokenAmount: string; // Result amount of toToken in minimal divisible units
-    protocols: unknown[]; // Route of the trade
-    tx: {
-        from: string; // Transactions will be sent from this address
-        to: string; // Transactions will be sent to our contract address
-        data: string; // Bytes of call data
-        value: string; // Amount of ETH (in wei) will be sent to the contract address
-        gas: number; // Estimated amount of the gas limit, increase this value by 25%
-        gasPrice: string; // Gas price in wei
-    };
+    protocols?: unknown[]; // Route of the trade
+    tx: SwapTxMeta;
+}
+
+export interface SwapTxMeta {
+    from: string; // Transactions will be sent from this address
+    to: string; // Transactions will be sent to our contract address
+    data: string; // Bytes of call data
+    value: string; // Amount of ETH (in wei) will be sent to the contract address
+    gas: number; // Estimated amount of the gas limit, increase this value by 25%
+    gasPrice: string; // Gas price in wei
 }
+
+export const OneInchService = {
+    parseQuoteParams({
+        fromToken,
+        toToken,
+        amount,
+    }: SwapQuoteParams): OneInchSwapQuoteParams {
```

Comment:
we won't receive zero address like this `0x0000000000000000000000000000000000000000`, right?

Reply:
It's not possible because token addresses come from the asset selection components which rely on tokens that already exist on the list, and it's currently impossible to add the null address as a token.

And for the native currency tokens we use the `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` address